### PR TITLE
Update OncePerRequestFilter to match with spring-web

### PR DESCRIPTION
**Pull request by bboyz269**
Call "getAlreadyFilteredAttributeName()" instead of direct usage if private "this.alreadyFilteredAttributeName"

Fix the error of the property value is null when proxied by CGLIB (bean from extending SessionRepositoryFilter)

### CODE REVIEW

**Comment by bboyz269** [comment]
I'm not sure what would be suffice for the test you mention but I'll look it up as soon as I get available time at a PC.On Jul 17, 2020 9:28 PM, Eleftheria Stein-Kousathana <notifications@github.com> wrote:
@eleftherias requested changes on this pull request.

Thanks for the PR @bboyz269!
Could you add a test demonstrating the bug?
Also, please add Closes gh-1658 to the commit message, so that the issue you reported is automatically closed when this is merged.

**Comment by rwinch** [comment]
I think a unit test demonstrating that a subclass of OncePerRequestFilter that has overridden getAlreadyFilteredAttributeName will still work is sufficient.

**Comment by bboyz269** [comment]
> I think a unit test demonstrating that a subclass of OncePerRequestFilter that has overridden getAlreadyFilteredAttributeName will still work is sufficient.

Not quite, it has to be a subclass being proxied by Spring AOP.

@eleftherias I add a new test class `OncePerRequestFilterAopTests`, would you check it out?

**Comment by eleftherias** [comment]
Thanks for the updates @bboyz269!

In the original issue that you reported, gh-1658, was the class also being proxied by Spring AOP?

**Comment by bboyz269** [comment]
> In the original issue that you reported, [gh-1658](https://github.com/spring-projects/spring-session/issues/1658), was the class also being proxied by Spring AOP?

Yes, it was being proxied by Spring AOP with CGLIb.

File: spring-session-core/src/test/java/org/springframework/session/web/http/OncePerRequestFilterAopTests.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2014-2019 the original author or authors.
```
Comment (eleftherias) [inline spring-session-core/src/test/java/org/springframework/session/web/http/OncePerRequestFilterAopTests.java:2]: The copyright should go up to 2020.

File: spring-session-core/src/test/java/org/springframework/session/web/http/OncePerRequestFilterAopTests.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			return Mockito.mock(SessionRepository.class);
+		}
+
+		@Bean
+		public SessionRepositoryFilter filter() {
+			return new SessionRepositoryFilter(sessionRepository());
+		}
+
+		@AfterReturning("execution(* SessionRepositoryFilter.doFilterInternal(..))")
+		public void doInternalFilterPoincut() {
```
Comment (eleftherias) [inline spring-session-core/src/test/java/org/springframework/session/web/http/OncePerRequestFilterAopTests.java:64]: Small typo
```suggestion
		public void doInternalFilterPointcut() {
```

Comment (bboyz269) [reply spring-session-core/src/test/java/org/springframework/session/web/http/OncePerRequestFilterAopTests.java]: Fixed as suggested.

File: spring-session-core/src/test/java/org/springframework/session/web/http/OncePerRequestFilterAopTests.java (hunk 1 above)
Comment (bboyz269) [reply spring-session-core/src/test/java/org/springframework/session/web/http/OncePerRequestFilterAopTests.java]: Fixed as suggested.

**Comment by eleftherias** [comment]
Thanks for the PR @bboyz269! This is now merged into master.

**Pull request merged**